app/models/facility.rb

```diff
@@ -28,4 +40,58 @@ class Facility < ApplicationRecord
   def report_on_patients
     registered_patients
   end
+
+
+  def self.parse_facilities(file_contents)
+    facilities = []
+    CSV.parse(file_contents, headers: true, converters: :strip_whitespace) do |row|
+      facility = {organization_name: row['organization'],
+                  facility_group_name: row['facility_group'],
+                  name: row['facility_name'],
+                  facility_type: row['facility_type'],
+                  street_address: row['street_address'],
+                  village_or_colony: row['village_or_colony'],
+                  district: row['district'],
+                  state: row['state'],
+                  country: row['country'],
+                  pin: row['pin'],
+                  latitude: row['latitude'],
+                  longitude: row['longitude'],
+                  import: true}
+      facilities << facility
+    end
+    facilities.delete_if { |facility| facility.except(:import).values.all?(&:blank?) }
+  end
+
+  def self.read_import_file(file)
```

Comment:
The model shoudn't be concerned with reading a file. 
Do you think it's better to create a FileUploadHelper?